### ISSUE TITLE
Create SwiftLint file

### ISSUE THREAD

**Issue by abbeyjackson** [CodeDoesGood/org#116]

**Comment by abbeyjackson** [comment]
@Pahitos how's this going?

**Comment by Pahitos** [comment]
Not sure... It's hard to test all the possible combinations of code 😅 
I'll commit what I've got so far soon.

**Comment by GrfxGuru** [comment]
@Pahitos if you need someone to try it out, I'd be happy to run some of my code through it. That should pretty much break anything ;)

**Comment by abbeyjackson** [comment]
@Pahitos @GrfxGuru Also I sent Pavel a message on slack about the rules and what can be changed :)

**Comment by abbeyjackson** [comment]
@Pahitos if this is

**Comment by GrfxGuru** [comment]
@Pahitos Is this complete and should be closed, or should it be in the in progress column?

**Comment by dimpiax** [comment]
So it's done or not?
We about swiftlint for Atom?

**Comment by Pahitos** [comment]
I was somehow missing these notifications, sorry about that. 

The only part that I couldn't do is the 3rd and 4th point of project specific [Empty Lines and Line Breaks](https://github.com/CodeDoesGood/iOS-Template#empty-lines-and-line-breaks) rule:
```
// Some code

if foo != nil {

    // several lines of code
}

// More code```

**Comment by Pahitos** [comment]
To me personally, this seems kind of vague and IMO something like this should be up to the programmer. But if someone can figure out how to make a rule that would satisfy this, that'd be fine too. 
There's another similar rule for [closures](https://github.com/CodeDoesGood/iOS-Template#closures) (3rd example), which is even harder to implement.

**Comment by dimpiax** [comment]
@Pahitos could you send you *lint* source file?

**Comment by Pahitos** [comment]
It's been committed to the template already:
https://github.com/CodeDoesGood/iOS-Template/blob/master/iOS-Template/CodeDoesGood-iOSTemplate/.swiftlint.yml

**Comment by dimpiax** [comment]
@Pahitos so this is not real-time checker?

**Comment by Pahitos** [comment]
@dimpiax not sure what you mean by implementing it in Atom 🤔 
It shows the errors in Xcode when you build the project. I was following [these](https://github.com/realm/SwiftLint#defining-custom-rules) instructions

**Comment by dimpiax** [comment]
@Pahitos https://github.com/realm/SwiftLint#atom

**Comment by Pahitos** [comment]
@dimpiax hmmm... Just installed this plugin and it works 🤔 
What does this have to do with iOS-Template though?

**Comment by dimpiax** [comment]
Just to be sure that is works.

**Comment by Pahitos** [comment]
Yup. As long as the plugin is installed, it should be all good 👍

**Comment by abbeyjackson** [comment]
Pavel we talked several times about the line breaks on Slack and I told you
it wasn't important :) but that if the Swiftlint default is different than
you need to edit the style guide to match Swiftlint.
-- 

Sincerely,

Abbey Jackson

604-600-1450 <javascript:void(0);>
https://github.com/abbeyjackson
http://ca.linkedin.com/in/earthabbey

**Comment by abbeyjackson** [comment]
Dima Swiftlint is an add-in for Xcode that ensures every project adheres to
certain code formatting guidelines and best practices.
-- 

Sincerely,

Abbey Jackson

604-600-1450 <javascript:void(0);>
https://github.com/abbeyjackson
http://ca.linkedin.com/in/earthabbey

**Comment by dimpiax** [comment]
@abbeyjackson I am about best practice to lint in realtime.

**Comment by abbeyjackson** [comment]
Swiftlint will give warnings and errors in Xcode at compile time. This is a
standard tool used in iOS development.
-- 

Sincerely,

Abbey Jackson

604-600-1450 <javascript:void(0);>
https://github.com/abbeyjackson
http://ca.linkedin.com/in/earthabbey
